gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java

```diff
@@ -112,19 +129,29 @@ public void refresh() {
             } else {
                 // search for events and compute them
                 logger.debug("Events synchronization");
-
-                List<Event> events = eventRepository.findByTimeFrame(lastRefreshAt - lastDelay, nextLastRefreshAt).toList().blockingGet();
-
-                if (events != null && !events.isEmpty()) {
-                    // Extract only the latest events by type and id
-                    Map<AbstractMap.SimpleEntry, Event> sortedEvents = events
-                            .stream()
-                            .collect(
-                                    toMap(
-                                            event -> new AbstractMap.SimpleEntry<>(event.getType(), event.getPayload().getId()),
-                                            event -> event, BinaryOperator.maxBy(comparing(Event::getCreatedAt)), LinkedHashMap::new));
-                    computeEvents(sortedEvents.values());
+                Single<List<Event>> eventProcessing = eventRepository.findByTimeFrame(lastRefreshAt - lastDelay, nextLastRefreshAt)
+                        .toList()
+                        .flatMap(events -> {
+                            if (!events.isEmpty()) {
+                                // Extract only the latest events by type and id
+                                Map<AbstractMap.SimpleEntry, Event> sortedEvents = events
+                                        .stream()
+                                        .collect(
+                                                toMap(
+                                                        event -> new AbstractMap.SimpleEntry<>(event.getType(), event.getPayload().getId()),
+                                                        event -> event, BinaryOperator.maxBy(comparing(Event::getCreatedAt)), LinkedHashMap::new));
+
+                                return Flowable.fromIterable(sortedEvents.values())
+                                        .flatMapCompletable(this::computeEvent, false, 1)
+                                        .toSingleDefault(events);
+                            }
+
+                            return Single.just(events);
+                        });
+                if (eventsTimeOut > 0) {
+                    eventProcessing = eventProcessing.timeout(eventsTimeOut, TimeUnit.MILLISECONDS);
                 }
+                eventProcessing.blockingGet();
```

Comment:
Why do you need this as you don't do anything with the return object? Shouldn't you use Completable instead?
I would change previous flatMap to flatMapCompletable; remove code line 146 and return Completable.complete line 149